osu.Game/Graphics/UserInterface/HistoryTextBox.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using osu.Framework.Input.Events;
+using osu.Game.Rulesets.Difficulty.Utils;
```

Comment:
As [mentioned in the previous review round](https://github.com/ppy/osu/pull/21228#discussion_r1024781895), `LimitedCapacityQueue` should be moved to a more general namespace if it's to be used here.

Reply:
Not sure where to move it, my guess would be `osu.Game.Utils`?

Reply:
Sounds about right.